File: vdp-ruby/test/visadirect/watch_list_screening_test.rb
require 'test/unit'
require 'rest-client'
require 'json'
require 'yaml'
require File.expand_path('../../../lib/visa_api_client', __FILE__)

class WatchListScreening < Test::Unit::TestCase
  def setup
    @visa_api_client = VisaAPIClient.new
    @watchListInquiry ='''{
    "acquirerCountryCode": "840",
    "acquiringBin": "408999",
    "address": {
      "city": "Bangalore",
      "cardIssuerCountryCode": "IND"
    },
    "referenceNumber": "430000367618",
    "name": "Mohammed Qasim"
  }'''
  end
  
  def test_watchListInquiry
    base_uri = 'visadirect/'
    resource_path = 'watchlistscreening/v1/watchlistinquiry'
    response_code = @visa_api_client.doMutualAuthRequest("#{base_uri}#{resource_path}", "Watch List Inquiry call test", "post", @watchListInquiry)
    assert_equal("200", response_code, "Watch List Inquiry test failed")
  end
end

File: visa-dev-samplecode-master/vdp-node/test/visadirect/watchlistscreeningtest.js
var request = require('request');
var fs = require('fs');
var config = require('../../config/configuration.json');
var VisaAPIClient = require('../../libs/visaapiclient.js');
var assert = require('chai').assert;
var randomstring = require('randomstring');

var req = request.defaults();
var userId = config.userId ;
var password = config.password;
var keyFile = config.key;
var certificateFile = config.cert;

describe('Visa Direct Watch List Screening test', function() {
	var visaAPIClient = new VisaAPIClient();
	var watchListInquiry = JSON.stringify({
		  "acquirerCountryCode": "840",
		  "acquiringBin": "408999",
		  "address": {
		    "city": "Bangalore",
		    "cardIssuerCountryCode": "IND"
		  },
		  "referenceNumber": "430000367618",
		  "name": "Mohammed Qasim"
		});
	
	it('Watch List Inquiry Test',function(done) {
		this.timeout(10000);
		var baseUri = 'visadirect/';
		var resourcePath = 'watchlistscreening/v1/watchlistinquiry';
		visaAPIClient.doMutualAuthRequest(baseUri + resourcePath, watchListInquiry, 'POST', {}, 
		function(err, responseCode) {
			if(!err) {
				assert.equal(responseCode, 200);
			} else {
				assert(false);
			}
		    done();
		});
	});
});
